src/base/searchengine.h

```diff
@@ -38,10 +40,133 @@
 class QProcess;
 class QTimer;
 
+template <class T, std::size_t N>
+class Version
+{
+public:
+    typedef T ComponentType;
+    typedef Version<T,N> this_type;
+
+    Version()
+    {
+        m_components.fill(T());
+    }
+
+    template <class ...Other>
+    Version(Other... components)
+        : m_components {{components...}}
+    {
+    }
+
+    Version(const QString &version)
```

Comment:
Maybe what I'm going to say doesn't make sense:
It is weird to have a `QString` parameter ctor. I would imagine having a list (or tuple) of numbers as parameters.